### PR TITLE
docs: add overview of key incompatibilities between KSQL and ksqlDB

### PR DESCRIPTION
**Pull request by derekjn**
Outlines key differences and incompatibilities between KSQL and ksqlDB.

### CODE REVIEW

File: docs-md/operate-and-deploy/ksql-vs-ksqldb.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+-----------------------
+
+In previous KSQL versions, push query responses could return multiple JSON objects that were not wrapped in an array. Thus a collection of objects returned by the server was not a valid JSON document. In ksqlDB, the HTTP response format for [push queries](/concepts/queries/push) has changed such that all responses are now valid JSON documents. Push query results returned by a ksqlDB server will be wrapped in a JSON array.
+
+Additionally, push query HTTP responses in ksqlDB will now include a query result schema as well as the ID assigned to the query. Finally, null fields are no longer included in push query result rows.
+
+Topic Name Casing
+--------------------
+
+Previous KSQL versions automatically uppercased all topics names created by the KSQL server. ksqlDB does not automatically uppercase the names of topics it creates. The casing of the named objects that ksqlDB derives topic names from (e.g. persistent queries) will be preserved.
```
Comment (vcrfxia) [inline docs-md/operate-and-deploy/ksql-vs-ksqldb.md]: Is this true? The name of the topic defaults to the name of the stream/table (potentially with a prefix, specified as a server config) which KSQL still converts to upper-case by default (unless the stream/table name is quoted).

Comment (derekjn) [reply docs-md/operate-and-deploy/ksql-vs-ksqldb.md]: @vcrfxia does this commit mean something different?

31b78a8d117e9567e085f0ba71a94ad928216fb4

Comment (vcrfxia) [reply docs-md/operate-and-deploy/ksql-vs-ksqldb.md]: That's only for the `PRINT TOPIC` command. Previously if a user submitted the statement `PRINT TOPIC foo;` KSQL would try to print the topic `FOO` but with that change KSQL now tries to print `foo`. However, if KSQL creates a topic, e.g. via `CREATE STREAM foo AS SELECT ...` then the topic name is still `FOO`.

cc @purplefox to confirm.

Comment (derekjn) [reply docs-md/operate-and-deploy/ksql-vs-ksqldb.md]: Ok, seems like this section should just be removed entirely then.

File: docs-md/operate-and-deploy/ksql-vs-ksqldb.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+!!! note
+    `EMIT CHANGES` will be required for persistent queries in the future. We
+    recommend migrating all of your persistent streaming queries to include
+    the EMIT CHANGES modifier.
+
+Storage
+--------
+
+ksqlDB introduces changes to the underlying materialization storage that are
```
Comment (vcrfxia) [inline docs-md/operate-and-deploy/ksql-vs-ksqldb.md]: I don't understand. What were the changes? I just spun up a 5.3 KSQL server, submitted some queries, then stood up ksqlDB from `master` and it came up fine. Are you saying that wasn't supposed to happen?

Comment (derekjn) [reply docs-md/operate-and-deploy/ksql-vs-ksqldb.md]: In general it's not expected to work. The test you did probably just happened to be compatible with `0.6.0`. @big-andy-coates can provide more details.

Comment (big-andy-coates) [reply docs-md/operate-and-deploy/ksql-vs-ksqldb.md]: All of the compatibility breaking config defs have been disabled in master (and ksqlDB).  So we can't guarantee any backward compatibility with running queries from previous versions.

It's possible queries started on 5.3 / 5.4 currently work with 0.6, but that's by chance, not by design.

**Pull request merged**